aten/src/ATen/native/cudnn/Conv.cpp

```diff
@@ -583,7 +603,18 @@ struct algorithm_search<cudnnConvolutionBwdDataAlgoPerf_t> {
           ws.data,
           ws.size));
     }
-    return getBestAlgorithm<perf_t>(perf_results.get(), args.params.deterministic, perf_count);
+    
+    // See Note [blacklist fft algorithms for strided dgrad]
+    int stride_dim = args.input.dim() - 2;
+    bool blacklist = false;
+    for (int i = 0; i< stride_dim; i++) {
+      if (args.params.stride[i] != 1) {
+          blacklist = true;
+          break;
+      }
+    }
+
+    return getBestAlgorithm<perf_t>(perf_results.get(), args, perf_count, blacklist);
```

Comment:
`blacklist` can be computed from `args`, so why can't we do this inside `getBestAlgorithm`? You can use templates to guard this code with an
```cpp
if (std::is_same<perf_t, cudnnConvolutionBwdDataAlgoPerf_t>::value) {
  // check if should blacklist
}
```

Reply:
Oh nice :). We could do that. 

CC: @ngimel